File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {

        int c = 0;
        int p = 1;
        vector<int> r(nums.size(),0);

        for(int i=0;i<nums.size();i++)
        {
            if(nums[i] == 0)
            {
                c++;
            }
            else
            {
                p = p*nums[i];
            }
        }

        if(c>1)
            return r;
        else if(c==1)
        {
            for(int i=0;i<nums.size();i++)
            {
                if(nums[i] == 0)
                    r[i] = p;
            }
        }
        else
        {
            for(int i=0;i<nums.size();i++)
            {
                r[i] = p/nums[i];
            }
        }
    return r;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums: number[]): number {
        const numSet = new Set(nums);
        let r = 0;

        for(const num of numSet){
            if(!numSet.has(num - 1)){
                let length = 1;
                while(numSet.has(num + length))
                {
                    length++;
                }
                r = Math.max(r, length);
            }
        }

        return r;

    }
}
